src/query_builder.ts

```diff
@@ -9,7 +13,12 @@ export default function interpolateKustoQuery(query: string, scopedVars?: Scoped
 
   query = query.replace(macroRegexp, (match, p1, p2) => {
```

Comment:
one last nit :sweat_smile: you can do:

```ts
return replace(query.replace ...
```

that way, functions calling `interpolateKustoQuery` do not need to call `templateSrv.replace` again, it's automatically done here.

Reply:
Do you mean replace the final `return` on line 34 with `return replace(query, ...)`? This would then mean we could remove lines 97 and 233 in `datasource.ts`

Reply:
yes, and that you don't need to redefine `query = query`, simply return the value

Reply:
I think we can't return immediately at this point as we still have the interval regex to run and replace any interval variables. Unless I'm missing something?

Reply:
(for some reason this message was not sent)

You are right!, I was looking at the github view that collapsed the other replacing :)